packages/core/code-runner/test/code-runner.spec.js

```diff
@@ -0,0 +1,53 @@
+import assert from 'assert';
+import sinon from 'sinon';
+import { fakeEmbark } from 'embark-testing';
+import CodeRunner from '../src/';
+
+// Due to our `DAPP_PATH` dependency in `embark-utils` `dappPath()`, we need to
+// ensure that this environment variable is defined.
+process.env.DAPP_PATH = 'something';
+
+describe('core/code-runner', () => {
+
+  const { embark, plugins } = fakeEmbark();
+
+  let codeRunner, doneCb;
+
+  beforeEach(() => {
+    codeRunner = new CodeRunner(embark);
+    doneCb = sinon.fake();
+  });
+
+  afterEach(() => {
+    embark.teardown();
+    sinon.restore();
+  });
+
+  test('it should register variables and eval code in the VM', done => {
+    const testVar = {
+      foo: 'bar'
+    };
+    embark.events.request('runcode:register', 'testVar', testVar, () => {
```

Comment:
can we simplify this with `await embark.events.request2(` etc.. ?

Reply:
@iurimatias we need to update `embark-testing` for that first.